examples/src/bin/uart_send_break.rs

```diff
@@ -0,0 +1,37 @@
+//! Example of sending a software break signal from a UART in
+//! Blocking mode.
+//!
+//! The following wiring is assumed:
+//! - TX => GPIO17
+//% CHIPS: esp32
+//% FEATURES: esp-hal/unstable
+
+#![no_std]
+#![no_main]
+
+use esp_backtrace as _;
+use esp_hal::{
+    delay::Delay,
+    main,
+    uart::{Config as UartConfig, DataBits, Parity, StopBits, Uart},
+};
+
+#[main]
+fn main() -> ! {
+    let peripherals = esp_hal::init(esp_hal::Config::default());
+    let uart_config = UartConfig::default()
+        .with_baudrate(19200)
+        .with_data_bits(DataBits::_8)
+        .with_parity(Parity::None)
+        .with_stop_bits(StopBits::_1);
+    let mut uart = Uart::new(peripherals.UART1, uart_config)
+        .expect("Failed to initialize UART")
+        .with_tx(peripherals.GPIO17);
+    let delay = Delay::new();
+
+    loop {
+        // Send a break signal for 20 bits
+        uart.send_break(20);
```

Comment:
This test is inadequate, we need something to check the signals coming from the UART TX line. We usually use PCNT or RMT for this purpose, check out some of the other tests.

Reply:
This is just for the `examples/` to demonstrate the usage, where would the test go?

Reply:
Ah sorry, I didn't quite release what I was looking at 😅 . Take a look at the `hil-test` package, writing tests there will run them on the esp's we have in our server room. The README in there should explain most things, and looking at the tests for inspiration should help too.

Reply:
When I fold the break interrupt PR into this, I'll HIL test the two together as well.